README
NAME
    MooseX::AttributeIndexes - Advertise metadata about your
    Model-Representing Classes to Any Database tool.

VERSION
    version 2.000001

SYNOPSIS
  Implementing Indexes
      package My::Package;
      use Moose;
      use MooseX::AttributeIndexes;
      use MooseX::Types::Moose qw( :all );

      has 'id' => (
        isa => Str,
        is  => 'rw',
        primary_index => 1,
      );

      has 'name' => (
        isa => Str,
        is  => 'rw',
        indexed => 1,
      );

      has 'foo' => (
        isa => Str,
        is  => 'rw',
      );

  Accessing Indexed Data
      package TestScript;

      use My::Package;

      my $foo = My::Package->new(
        id => "Bob",
        name => "Smith",
        foo  => "Bar",
      );

      $foo->attribute_indexes
      # { id => 'Bob', name => 'Smith' }

  Using With Search::GIN::Extract::Callback
      Search::GIN::Extract::Callback(
        extract => sub {
          my ( $obj, $callback, $args ) = @_;
          if( $obj->does( 'MooseX::AttributeIndexes::Provider') ){
            return $obj->attribute_indexes;
          }
        }
      );

  CODERef
    Since 0.01001007, the following notation is also supported:

      has 'name' => (
        ...
        indexed => sub {
          my ( $attribute_meta, $object, $value ) = @_;
          return "$_" ; # $_ == $value
        }
      );

    Noting of course, $value is populated by the meta-accessor.

    This is a simple way to add exceptions for weird cases for things you
    want to index that don't behave like they should.

   SEE ALSO
    "Search::GIN::Extract::AttributeIndexes"

AUTHORS
    *   Kent Fredric <kentnl@cpan.org>

    *   Jesse Luehrs <doy@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by Kent Fredric
    <kentfredric@gmail.com>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for MooseX-AttributeIndexes

2.000002 2017-03-09T05:33:17Z 049cdd6

2.000001 2017-03-09T05:26:26Z a63676f
 - Bugfix: Avoid test failures with -Ddefault_inc_excludes_dot

 [Dependencies::Stats]
 - Dependencies changed since 2.000000, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +13 ↑4 -3 (recommends: +1, suggests: ↑2)
 - runtime: (recommends: +1)
 - test: +1 (recommends: ↑1 ↓1)

2.000000 2014-09-24T05:26:53Z
 [00 Major - Toolkit]
 - This may break a hand-operated downstream.
 - Migrated from using Module::Build as an installer to using EUMM
 - "Latest" dependencies softened to recommendations instead of requires.
 - https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f

 [00 Major - Version Scheme]
 - This may break a hand-operated downstream.
 - Migrated from 1.x.y to 2.xxxyyy
 - This is the primary reason for the 2.x on the box.
 - https://questhub.io/realm/perl/quest/5388d3a9bbd0be2224000129

 [Dependencies::Stats]
 - Dependencies changed since 1.0.3, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +46 -2 (recommends: -1, suggests: +1 ↑1)
 - runtime: +1 ↓1 -1
 - test: +3 ↓1 -2 (recommends: +3)

 [Internals]
 - use namespace::clean instead of autoclean.

 [Metadata]
 - Add email address to copyright.
 - Recognise 2x Contributors.
 - Migrate to github/kentnl

 [Tests]
 - Use Test::Compile::PerFile instead of Test::Compile
 - Use Test::ReportPrereqs instead of Test::ReportVersions::Tiny

1.0.3 2013-04-06T05:09:44Z
 - Maintenance release for Module::Build 0.4004

 [Dependencies::Stats]
 - Dependencies changed since 1.0.2, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +5 (recommends: ↑1, suggests: ↑1)
 - test: ↑1

 [Documentation]
 - update copyright year
 - reindent license
 - update license address
 - add README.mkdn

 [Meta]
 - bugtracker to github issues

 [Packaging]
 - ship perltidyrc
 - utilise test_requires in Build.PL

1.0.2 2011-10-30T19:28:41Z
 - Maintenance Release

 [Dependencies::Stats]
 - Dependencies changed since 1.0.1, see misc/*.deps* for details
 - develop: (suggests: ↑1)
 - runtime: +3
 - test: -1

 [Internals]
 - $VERSION set outside begin
 - $AUTHORITY now defined
 - code tidied

 [Packaging]
 - Update License ( Indent, Address )
 - Move author/release tests to xt/
 - Repository urls moved to https://
 - added x_authority

1.0.1 2011-04-04T07:40:17Z
 [Dependencies::Stats]
 - Dependencies changed since 1.0.0, see misc/*.deps* for details
 - develop: +1 -1 (recommends: +1 -1, suggests: +1 -1)

 [Features]
 - Rework code to to work properly with role attributes. ( Thanks doy / Jesse Luehrs )

 [Packaging]
 - Rework Changelog for CPAN::Changes 1:1 compatibility.
 - Move from @KENTNL to @Author::KENTNL

1.0.0 2011-01-23T04:01:35Z
 [BugFix]
 - Fixed MOP deprecation warning with CMOP 0.95 ( Thanks Gerhard Gossen )
 - Stop using deprecated get_method_map.

 [Dependencies::Stats]
 - Dependencies changed since 0.01001007, see misc/*.deps* for details
 - build: +1
 - configure: +1
 - develop: +1 (recommends: +1, suggests: +1)
 - runtime: +3 ↑1 ↓1 -3
 - test: +5

 [Features]
 - Migrate to new MetaRole API.

 [Packaging]
 - Migrate to 3.dot version scheme.
 - Update toolchain.
 - Ship dist.ini
 - CompileTests

0.01001007 2009-08-31T19:52:09Z
 - Documented that feature.
 - Added support for per-attribute callback processors

 [Dependencies::Stats]
 - Dependencies changed since 0.01000613, see misc/*.deps* for details
 - runtime: +1

0.01000613 2009-08-28T01:55:48Z
 - Ramped up deps to hopefully solve weird Mx::Types Carp::Clan failure.

 [Dependencies::Stats]
 - Dependencies changed since 0.01000404, see misc/*.deps* for details
 - runtime: +1 ↑2

0.01000404 2009-08-25T16:18:59Z
 - Depend Resolution Fixup
 - See-Also doc,
 - Perltidy

 [Dependencies::Stats]
 - Dependencies changed since 0.01000300, see misc/*.deps* for details
 - runtime: +1

0.01000300 2009-08-24T12:06:14Z
 - First version.


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4210"
  },
  "configure_requires" => {
    "Module::Build" => "0.4210"
  },
  "dist_abstract" => "Advertise metadata about your Model-Representing Classes to Any Database tool.",
  "dist_author" => [
    "Kent Fredric <kentnl\@cpan.org>",
    "Jesse Luehrs <doy\@cpan.org>"
  ],
  "dist_name" => "MooseX-AttributeIndexes",
  "dist_version" => "2.000000",
  "license" => "perl",
  "module_name" => "MooseX::AttributeIndexes",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Moose" => "0.94",
    "Moose::Exporter" => 0,
    "Moose::Role" => 0,
    "Moose::Util::MetaRole" => 0,
    "MooseX::Types::Moose" => "0.19",
    "Scalar::Util" => 0,
    "namespace::autoclean" => "0.08",
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::Moose" => 0,
    "Test::More" => "1.001006",
    "lib" => 0
  }
);


my %fallback_build_requires = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Build" => "0.4210",
  "Test::Moose" => 0,
  "Test::More" => "1.001006",
  "lib" => 0
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
